src/server/session.ts

```diff
@@ -769,6 +769,22 @@ namespace ts.server {
             return project.getLanguageService().getRenameInfo(file, position);
         }
 
+        private isValidIdentifierName(args: protocol.ValidateIdentifierNameArgs): IsValidIdentifierName {
+            const project = this.getProject(args.projectFileName);
+
+            if (ts.isKeyword(stringToToken(args.identifierName)) || !ts.isIdentifierText(args.identifierName, project.getCompilerOptions().target)) {

```

Comment:
what if the location of the rename allows for non-identifier names, e.g. inside a string literal?

Reply:
In that case the string will be reported as an error. In my opinion, the service client should be able to handle this scenarios. This method will only report if the string is a valid identifier or not regardless of it's location.